### PR TITLE
XL/fs: Initialize export paths supplied on command line

### LINKED ISSUE #2013: XL/FS: Minio server dir - auto create if dir is missing
## Minio version
- (paste output of `minio version`)

### PR DESCRIPTION
**Pull request by harshavardhana**
Fixes #2013

### CODE REVIEW

File: xl-v1.go [hunk 1]
```diff
@@ -120,6 +121,12 @@ func newXLObjects(disks []string) (ObjectLayer, error) {
 		if err != nil && err != errDiskNotFound {
 			return nil, err
 		}
+		// Disk not found create it.
+		if err == errDiskNotFound {
+			if err = os.MkdirAll(disk, 0777); err != nil {
```
Comment (krishnasrinivas) [inline xl-v1.go]: this has to go through StorageAPI if we take into account multinode version

**Comment by balamurugana** [comment]
Can you change commit log with more appropriate?

**Comment by harshavardhana** [comment]
> Can you change commit log with more appropriate?

How does it look now?

File: fs-v1.go [hunk 2]
```diff
@@ -81,9 +81,15 @@ func shutdownFS(storage StorageAPI) {
 // newFSObjects - initialize new fs object layer.
 func newFSObjects(disk string) (ObjectLayer, error) {
 	storage, err := newStorageAPI(disk)
-	if err != nil {
+	if err != nil && err != errDiskNotFound {
 		return nil, err
 	}
+	// Disk not found create it.
+	if err == errDiskNotFound {
+		if err = os.MkdirAll(disk, 0777); err != nil {
```
Comment (abperiasamy) [inline fs-v1.go]: It will be good to verify this with a unit test as well.

File: test-utils_test.go [hunk 3]
```diff
@@ -61,6 +63,29 @@ const (
...
+// TempFile to a minimum.
+var randN uint32
+var randmu sync.Mutex
+
+func reseed() uint32 {
+	return uint32(time.Now().UnixNano() + int64(os.Getpid()))
+}
+
+func nextSuffix() string {
+	randmu.Lock()
```
Comment (abperiasamy) [inline test-utils_test.go:79]: require func comments

**Pull request merged**